robocop/checkers/misc.py

```diff
@@ -1725,3 +1870,83 @@ def check_order(self, node):
                 max_order_indicator = this_node_expected_order
 
     visit_Keyword = visit_TestCase = check_order  # noqa: N815
+
+
+class NonLocalVariableChecker(VisitorChecker):
+    reports = (
+        "no-global-variable",
+        "no-suite-variable",
+        "no-test-variable",
+    )
+
+    def visit_KeywordCall(self, node: KeywordCall):  # noqa: N802
+        keyword_token = node.get_token(Token.KEYWORD)
+
+        if not keyword_token:
+            return
+
+        keyword_name = normalize_robot_name(keyword_token.value, remove_prefix="builtin.")
+
```

Comment:
We could speed up the look up for 'invalid' names here by grouping them under single set:

```
GLOBAL_VARIANTS = {'setglobalvariable', ..}

(...)

    if keyword_name not in self.GLOBAL_VARIANTS:  # O(1) look up time, and most keywords with return here
        return
    # rest of the existing code here
```

Reply:
I made the set in the checker class. I think moving it outside the class will cause readability issues.

Reply:
Good, I didnt outline it but it was supposed to be in the checker class so good choice

Reply:
I assumed you meant a global variable due to the `SCREAMING_SNAKE_CASE` syntax you used :)